server/reports/README.md

```diff
@@ -0,0 +1,451 @@
+Reports of a new patch must be backwards compatible to open mSupply of the same major and minor version. Therefore we can reduce PR dif overhead, and committed file overhead in OMS, by editing a report if it is a patch change.
+
+Previous patch versions of a report can be accessed if necessary on old branches of open mSupply.
+
+#### Major or minor version change
+
+A copy of the source files must be made for major or minor report version changes.
+
+These changes should only be made when the api of the report is changed, causing it to no longer be compatible with the older version of open mSupply.
+
+In these cases, it is necessary to produce both versions of the report to ensure a report is available for remote OMS sites which might not have been updated to the latest version.
+
+Convention for file structures of report versions is:
+
+```
+└──  example-report
+   ├── 2_4
+   └── 2_5
+```
+
+Where 2_4 and 2_5 are directories containing source files of different versions of the same report.
+
+### File Structure
+
+#### Reports and Forms
+
+Reports are separated into 'reports' and 'forms' (previously picklists)
+
+##### Reports
+
+Use argument schemas to generate filtered reports based on user input.
+
+##### Forms
+
+Use inbuilt constant arguments, and are not customisable with front end user inputs.
+
+##### File Structure
+
+The full conventional file structure is as follows:
+
+```
+├── reports
+   ├──  clients
+   |   ├── client 1
+   |   |   ├── Reports
+   |   |   |   ├── expiring-items
+   |   |   |   |   ├── 2_6
+   |   |   |   |   └── 2_7
+   |   |   |   └── item-usage
+   |   |   |       └── 3_2
+   |   |   └── Forms
+   |   |       ├── repack
+   |   |       |   ├── 2_6
+   |   |       |   └── 2_7
+   |   |       └── requisition
+   |   |           └── 2_7 
+   |   └── client 2
+   |       └── Reports
+   |           └── expiring-items
+   |               ├── 2_6
+   |               └── 2_7
+   └── Other source files and helper functions
+```
+
+Where each 'version dir' ie 2_6 contains all the source files required to build that report as per the [source file diagram](#source-file-structure-diagram)
+
+### Client Confidentiality
+
+Client specific reports are confidential. Information regarding these reports must be kept out of the open-msupply repository.
+
+When sharing databases, ensure no client information is uploaded github. Share databases and client specific information through secure channels only.
+
+## Maintenance (support info)
+
+The OMS CLI provides tools for managing and developing reports within an omSupply instance. It allows users to build, upsert, and test reports, streamlining the development and deployment process.
+
+All sites will have embedded standard reports inserted on startup.
+
+Additional reports (both custom, and patched standard reports) can be upserted in bulk to open mSupply central servers via the `upsert-reports` command.
+
+Reports will then be synced out from there to remote sites.
```

Comment:
Sync is not yet developed, but is planned to be shortly.

I wasn't sure if this should be part of these changes